lib/iris/_concatenate.py

```diff
@@ -115,9 +115,9 @@ def __new__(mcs, coord, dims):
             kwargs["circular"] = coord.circular
         if isinstance(coord, iris.coords.DimCoord):
             # Mix the monotonic ordering into the metadata.
-            if coord.core_points()[0] == coord.core_points()[-1]:
+            if coord.points[0] == coord.points[-1]:
                 order = _CONSTANT
-            elif coord.core_points()[-1] > coord.core_points()[0]:
+            elif coord.points[-1] > coord.points[0]:
```

Comment:
We're exclusively dealing with `DimCoord`s here. Therefore the points and bounds are **never** lazy, so there is no need to use `core_points` to access the realized data.